src/core/src/op/constant.cpp

```diff
@@ -326,7 +326,9 @@ std::string ov::op::v0::Constant::convert_value_to_string(size_t index) const {
 }
 
 size_t ov::op::v0::Constant::get_byte_size() const {
-    return m_data->size();
+    // Returns 0 when shape is "empty" (equals 0).
+    // TODO: refactor shape_size(m_shape) calculations and store it as a member.
+    return shape_size(m_shape) ? m_data->size() : 0;
 }
```

Comment:
@praasz @jane-intel @ilya-lavrenov here `get_byte_size()` is currently returning `1` due to the internal implementation of Core. As for "empty data of shape [0]" expected value should be `0` as well -- there are no bytes. So I have edited this function to be based on `shape_size(m_shape)` instead of `m_data` itself.

Can we all agree on this new behavior? It seems like the "0-bytes" case was not considered before but it is allowed. Or it shouldn't be allowed at all?

There is also a note about future improvement of performing `shape_size(m_shape)` calculation only one-time as it is recurring call and . However this should be a separate PR.

Reply:
For me if there will be no regression in tests/models etc. this change is OK.

In general, I think the `get_byte_size()` function should return same value for `ov::Tensor` or `Constant` if both classes use same shape and precision.